modules/server/src/topics/actionkeybindtopic.cpp

```diff
@@ -0,0 +1,149 @@
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2025                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/server/include/topics/shortcuttopic.h>
+
+#include <modules/server/include/connection.h>
+#include <openspace/engine/globals.h>
+#include <openspace/interaction/actionmanager.h>
+#include <openspace/interaction/keybindingmanager.h>
+#include <ghoul/logging/logmanager.h>
+
+using nlohmann::json;
+
+namespace openspace {
+
+bool ActionKeybindTopic::isDone() const {
+    return true;
+}
+
+nlohmann::json jsonAction(const interaction::Action& action) {
+    // Convert to std as nlohmann doesn't understand glm
+    std::vector<float> colorVec;
+    if (action.color.has_value()) {
+        glm::vec4 color = action.color.value();
+        colorVec = { color.r, color.g, color.b, color.a };
+    }
+
+    const nlohmann::json shortcutJson = {
+        { "identifier", action.identifier },
+        { "name", action.name },
+        { "synchronization", static_cast<bool>(!action.isLocal) },
+        { "documentation", action.documentation },
+        { "guiPath", action.guiPath },
+        { "color", colorVec }
```

Comment:
now we would send an empty array if there is no color. Might be better to not include it at all in that case, maybe?  Easier to deal with undefined/null than an empty array and it makes it more clear when an action has a color or not